internal/csi-common/nodeserver-default.go

```diff
@@ -112,6 +114,53 @@ func (ns *DefaultNodeServer) NodeGetVolumeStats(ctx context.Context, req *csi.No
 
 	*/
 
+	stat, err := os.Stat(targetPath)
+	if err != nil {
+		return nil, status.Errorf(codes.InvalidArgument, "failed to get stat for targetpath %q: %v", targetPath, err)
+	}
+
+	if stat.Mode().IsDir() {
+		return filesystemNodeGetVolumeStats(ctx, targetPath)
+	} else if (stat.Mode() & os.ModeDevice) == os.ModeDevice {
+		return blockNodeGetVolumeStats(ctx, targetPath)
+	}
+
+	return nil, fmt.Errorf("targetpath %q is not a directory or device", targetPath)
+}
+
+// blockNodeGetVolumeStats gets the metrics for a `volumeMode: Block` type of
+// volume. At the moment, only the size of the block-device can be returned, as
+// there are no secrets in the NodeGetVolumeStats request that enables us to
+// connect to the Ceph cluster.
+//
+// TODO: https://github.com/container-storage-interface/spec/issues/371#issuecomment-756834471
+func blockNodeGetVolumeStats(ctx context.Context, targetPath string) (*csi.NodeGetVolumeStatsResponse, error) {
+	args := []string{"--noheadings", "--bytes", "--output=SIZE", targetPath}
+	lsblkSize, _, err := util.ExecCommand(ctx, "/bin/lsblk", args...)
```

Comment:
validate stdErr also for the safer side?

Reply:
We could, not sure if it is useful enough. The `err` should contain sufficient details already.

Reply:
sometimes the err contains error codes but the stderr contains the human-readable error message.